Content.IntegrationTests/Tests/Weapons/WeaponSniperTests.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Content.IntegrationTests.Tests.Interaction;
+using Content.Shared.Damage;
+using Content.Shared.Weapons.Ranged.Components;
+using Content.Shared.Weapons.Ranged.Events;
+using Content.Shared.Wieldable.Components;
+using Robust.Shared.Prototypes;
+
+namespace Content.IntegrationTests.Tests.Weapons;
+
+public sealed class WeaponSniperTests : InteractionTest
+{
+    private static readonly EntProtoId MobHuman = "MobHuman";
+    private static readonly EntProtoId SniperMosin = "WeaponSniperMosin";
+
+    [Test]
+    public async Task SniperMosinInteractionTest()
```

Comment:
```suggestion
    public async Task GunRequiresWieldTest()
```
We are testing components, not this prototype in particular, so it would be better to name it after that.
Also leave a code comment what this is testing, like "Tests that a gun with GunRequiresWieldComponent can only be shot when wielded"